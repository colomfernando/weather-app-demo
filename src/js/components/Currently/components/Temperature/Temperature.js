import React from 'react';
import { number } from 'prop-types';
import { validateNumber } from 'core/utils';
import Styles from './styles';

const Temperature = ({ temperature, feelLike }) => {
	if (!validateNumber(temperature) || !validateNumber(feelLike)) return null;
	return (
		<Styles.Wrapper>
			<Styles.TemperatureText>{`${Number.parseInt(temperature, 10)}ยบ`}</Styles.TemperatureText>
			<Styles.WrapperFeel>
				<Styles.FeelLikeText>Feel like</Styles.FeelLikeText>
				<Styles.FeelLikeText>{`${Number.parseInt(feelLike, 10)}ยบ`}</Styles.FeelLikeText>
			</Styles.WrapperFeel>
		</Styles.Wrapper>
	);
};

Temperature.propTypes = {
	temperature: number.isRequired,
	feelLike: number.isRequired
};

export default Temperature;
