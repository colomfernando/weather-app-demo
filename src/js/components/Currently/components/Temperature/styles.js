import styled from 'styled-components';
import { colors } from 'theme';

const baseText = styled.p`
	color: ${colors.white};
	font-size: 16px;
`;

const Wrapper = styled.div`
	display: flex;
	flex-flow: column nowrap;
`;

const WrapperFeel = styled.div`
	display: flex;
	justify-content: space-between;
`;

const TemperatureText = styled(baseText)`
	font-size: 80px;
	margin-bottom: 10px;
`;

const FeelLikeText = styled(baseText)`
	margin-bottom: 10px;
`;

export default { Wrapper, WrapperFeel, TemperatureText, FeelLikeText };
