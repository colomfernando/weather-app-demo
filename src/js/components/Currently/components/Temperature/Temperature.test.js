import React from 'react';
import { mount } from 'enzyme';
import Temperature from './Temperature';

describe('component', () => {
	it('match snapshot ', () => {
		const wrapper = mount(<Temperature temperature={20} feelLike={21} />);
		expect(wrapper).toMatchSnapshot();
	});

	// it('render ok ', () => {
	// 	const wrapper = mount(<Temperature temperature={20} feelLike={21} />);
	// 	expect(wrapper.props().temperature).toEqual(20);
	// 	expect(wrapper.props().feelLike).toEqual(21);
	// });

	// it('expect two paragraph', () => {
	// 	const wrapper = mount(<Temperature temperature={20} feelLike={21} />);
	// 	const text = wrapper.find('p');
	// 	expect(text.length).toBe(2);
	// });
	// it('value of texts', () => {
	// 	const wrapper = mount(<Temperature temperature={24} feelLike={32} />);
	// 	const temperatureText = wrapper
	// 		.find('div')
	// 		.childAt(0)
	// 		.text();
	// 	const feelLikeText = wrapper
	// 		.find('div')
	// 		.childAt(1)
	// 		.text();
	// 	expect(temperatureText).toEqual('24ยบ');
	// 	expect(feelLikeText).toEqual('Feel like: 32ยบ');
	// });
});
