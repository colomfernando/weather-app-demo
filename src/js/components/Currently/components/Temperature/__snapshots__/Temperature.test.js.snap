// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component match snapshot  1`] = `
<Temperature
  feelLike={21}
  temperature={20}
>
  <styles__Wrapper>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "styles__Wrapper-qxuzqr-1",
            "isStatic": false,
            "lastClassName": "grKBGo",
            "rules": Array [
              "display:flex;flex-flow:column nowrap;",
            ],
          },
          "displayName": "styles__Wrapper",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "styles__Wrapper-qxuzqr-1",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="styles__Wrapper-qxuzqr-1 grKBGo"
      >
        <styles__TemperatureText>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "styles__TemperatureText-qxuzqr-3",
                  "isStatic": false,
                  "lastClassName": "brcPgZ",
                  "rules": Array [
                    "color:",
                    "#fff",
                    ";font-size:16px;",
                    "font-size:80px;margin-bottom:10px;",
                  ],
                },
                "displayName": "styles__TemperatureText",
                "foldedComponentIds": Array [
                  "styles__baseText-qxuzqr-0",
                ],
                "render": [Function],
                "styledComponentId": "styles__TemperatureText-qxuzqr-3",
                "target": "p",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <p
              className="styles__baseText-qxuzqr-0 styles__TemperatureText-qxuzqr-3 brcPgZ"
            >
              20ยบ
            </p>
          </StyledComponent>
        </styles__TemperatureText>
        <styles__WrapperFeel>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "styles__WrapperFeel-qxuzqr-2",
                  "isStatic": false,
                  "lastClassName": "kkidPH",
                  "rules": Array [
                    "display:flex;justify-content:space-between;",
                  ],
                },
                "displayName": "styles__WrapperFeel",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "styles__WrapperFeel-qxuzqr-2",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="styles__WrapperFeel-qxuzqr-2 kkidPH"
            >
              <styles__FeelLikeText>
                <StyledComponent
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "styles__FeelLikeText-qxuzqr-4",
                        "isStatic": false,
                        "lastClassName": "HkUZh",
                        "rules": Array [
                          "color:",
                          "#fff",
                          ";font-size:16px;",
                          "margin-bottom:10px;",
                        ],
                      },
                      "displayName": "styles__FeelLikeText",
                      "foldedComponentIds": Array [
                        "styles__baseText-qxuzqr-0",
                      ],
                      "render": [Function],
                      "styledComponentId": "styles__FeelLikeText-qxuzqr-4",
                      "target": "p",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                >
                  <p
                    className="styles__baseText-qxuzqr-0 styles__FeelLikeText-qxuzqr-4 HkUZh"
                  >
                    Feel like
                  </p>
                </StyledComponent>
              </styles__FeelLikeText>
              <styles__FeelLikeText>
                <StyledComponent
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "styles__FeelLikeText-qxuzqr-4",
                        "isStatic": false,
                        "lastClassName": "HkUZh",
                        "rules": Array [
                          "color:",
                          "#fff",
                          ";font-size:16px;",
                          "margin-bottom:10px;",
                        ],
                      },
                      "displayName": "styles__FeelLikeText",
                      "foldedComponentIds": Array [
                        "styles__baseText-qxuzqr-0",
                      ],
                      "render": [Function],
                      "styledComponentId": "styles__FeelLikeText-qxuzqr-4",
                      "target": "p",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                >
                  <p
                    className="styles__baseText-qxuzqr-0 styles__FeelLikeText-qxuzqr-4 HkUZh"
                  >
                    21ยบ
                  </p>
                </StyledComponent>
              </styles__FeelLikeText>
            </div>
          </StyledComponent>
        </styles__WrapperFeel>
      </div>
    </StyledComponent>
  </styles__Wrapper>
</Temperature>
`;
